### ISSUE TITLE
Exercise 6... no looping allowed?

### ISSUE THREAD

**Issue by maxscott** [stackgl/shader-school#104]
I understand if the goal is to get the total-n00b (me :point_left:) to write a solution without any loops, but it isn't terribly clear that you can't do that based on the previous lessons and the exercises instructions. I got a message about using a for loop without a "constant" because it relies on the param, and a message that while's were 'not allowed' or something to that effect. I remember recursion was out, but I tried it just for kicks.

If there are such contraints in general, maybe during the intro to looping these constraints could be made known?

**Comment by mwalczyk** [comment]
Yeah I agree with this. A little further explanation here would've helped.

**Comment by warren-lockhart** [comment]
Wouldn't it be better to just make an exercise which squares a matrix, rather than forcing the student to do all sorts of ugly decision making due to the fact that looping and recursion aren't possible? It's also unlikely that you would raise a matrix to a power like this, you would likely use a diagonalization method or use a matrix library to do it, and then pass the new matrix to the shader. A function which squares a matrix is sufficient to demonstrate matrix multiplication, or you could create a function which multiplies a vector by a matrix, which was in the lesson and is highly relevant to graphics programming.

**Comment by sparkmorry** [comment]
I am having the same problem, though n is given 1<n<16, I can't figure out a way to tell the machine = =

**Comment by ambethia** [comment]
@sparkmorry I solved it like this:

```glsl
